src/Authenticator/TokenAuthenticator.php

```diff
@@ -98,11 +98,11 @@ protected function getTokenFromQuery(ServerRequestInterface $request, $queryPara
     {
         $queryParams = $request->getQueryParams();
 
-        if (isset($queryParams[$queryParam])) {
-            return $queryParams[$queryParam];
+        if (!array_key_exists($queryParam, $queryParams)) {
```

Comment:
why array_key_exists? It is either non-null or irrelevant IMO.

Reply:
Well I thought it doesn't matter what's inside, null, 0, '0', '' - all ok, and middleware can do that (mine casts to real types for API things) a token with `(int)0` should be legit.

Let me ask the other way around: What's the benefit of using isset here which does all that autocasting? Does array_key_exists hurt?

Reply:
isset doesnt do any auto-cast. array_key_exists() usually goes a) too fur, b) is 10-100x slower.

Reply:
I still want to see the performance difference on 7.2/7.1 but going to remove this now.